core/arch/arm/kernel/boot.c

```diff
@@ -118,30 +118,34 @@ __weak unsigned long plat_get_aslr_seed(void)
 	return 0;
 }
 
-#if defined(_CFG_CORE_STACK_PROTECTOR)
+#if defined(_CFG_CORE_STACK_PROTECTOR) || defined(CFG_WITH_STACK_CANARIES)
 /* Generate random stack canary value on boot up */
-__weak uintptr_t plat_get_random_stack_canary(void)
+__weak void plat_get_random_stack_canaries(void *buf, size_t blen, size_t clen)
 {
-	uintptr_t canary = 0xbaaaad00;
 	TEE_Result ret = TEE_ERROR_GENERIC;
 
+	assert(buf && clen && blen >= clen && !(blen % clen));
+
 	/*
 	 * With virtualization the RNG is not initialized in Nexus core.
 	 * Need to override with platform specific implementation.
 	 */
 	if (IS_ENABLED(CFG_NS_VIRTUALIZATION)) {
 		IMSG("WARNING: Using fixed value for stack canary");
-		return canary;
+		memset(buf, 0xab, blen);
+		goto exit;
 	}
 
-	ret = crypto_rng_read(&canary, sizeof(canary));
+	ret = crypto_rng_read(buf, blen);
 	if (ret != TEE_SUCCESS)
 		panic("Failed to generate random stack canary");
 
+exit:
 	/* Leave null byte in canary to prevent string base exploit */
-	return canary & ~0xffUL;
+	for (size_t i = 0; i < blen / clen; i++)
```

Comment:
Please declare `i` at the start of a block. Variables should not be declared mixed with code.